radixdlt/src/main/java/com/radixdlt/consensus/VertexStore.java

```diff
@@ -84,13 +90,23 @@ public void insertVertex(Vertex vertex) throws VertexInsertionException {
 	}
 
 	public Vertex commitVertex(Hash vertexId) {
-		Vertex vertex = vertices.get(vertexId);
-		if (vertex == null) {
+		final Vertex tipVertex = vertices.get(vertexId);
+		if (tipVertex == null) {
 			throw new IllegalStateException("Committing a vertex which was never inserted: " + vertexId);
 		}
+		Vertex vertex = tipVertex;
+		while (vertex != null && !committedVertices.containsKey(vertex.getId())) {
+			committedVertices.put(vertexId, vertex);
+			vertex = vertices.get(vertex.getParentId());
+		}
+
+		lastCommittedVertex.onNext(tipVertex);
```

Comment:
This is where we would/will also be doing some cleanup of no longer committable branches, right?

Reply:
yes